Add a page about versioning policy

ephys opened the pull request:
This PR adds a page that regroups the information about which version of Sequelize is compatible with what.

It is partially inspired by https://github.com/sequelize/meetings/issues/18, but doesn't include the 'one major release per year' part as that part is still being discussed.

I think this page will be very useful both for users, and us, to know what should be supported for a given release.

I also re-organised the 'dialect specific things' page to use the same dialect order as the rest of our pages (postgres & its variants, mariadb, mysql, mssql, sqlite, snowflake, db2, ibmi).
I plan on merging part of 'dialect specific things' with the 'getting started' page (to have a 'getting started' guide per dialect)

ephys commented:
According to the 'upgrade to v7', Sequelize 6 is supposed to support mssql 2012, but it's not testing against that

WikiRik commented:
> According to the 'upgrade to v7', Sequelize 6 is supposed to support mssql 2012, but it's not testing against that

Yeah, that's because they do not have Docker images for 2012. We can try to set it up in a different way, but no real priority on that one.

I'll review this in the coming days

ephys commented:
No, it's EOL. As long as no one complains about it, let's ignore the issue

WikiRik commented on src/pages/releases.md:
```diff
@@ -0,0 +1,175 @@
+
+## Db2 for IBM i Support Table
+
+*Db2 for IBM i* requires the use of the [odbc] npm package.  
```
Why is this italic?

ephys replied:
Initially because it's long and I wanted to emphasize that this was the whole name of the dialect, but it's not very useful and I can remove it for consistency

ephys replied:
Italic removed

WikiRik reviewed (commented):
In general this looks fine but the page is not showing up in the sidebar (or at least I couldn't directly get to it on mobile). Before releasing v7 we should test on more database versions since we mention support for versions => x while we do not test on them. But I will take that up soon

ephys commented:
> the page is not showing up in the sidebar

Yup, pages don't have sidebars. I've added it to the footer instead and added links to it in 'dialect specific things'

WikiRik commented on docs/other-topics/dialect-specific-things.md:
```diff
@@ -160,38 +138,66 @@ const sequelize = new Sequelize('database', null, null, {
+:::caution
 
-When passing options to the constructor, the concept of `database` is mapped to the ODBC `DSN`. You can provide additional connection string options to Sequelize using the `dialectOptions.odbcConnectionString`. This connection string is then appended with the values found in the `database`, `username`, and `password` parameters:
+sqlite3 has not been released since february 2021, and has security vulnerabilities and installation issues on newer platforms.
```
```suggestion
sqlite3 has not been released since February 2021, and has security vulnerabilities and installation issues on newer platforms.
```

WikiRik commented on src/pages/releases.md:
```diff
@@ -0,0 +1,175 @@
+
+| Sequelize                       | [Node.js][node-releases] | [Typescript][ts-releases] | Release Date | EOL        |
+|---------------------------------|--------------------------|---------------------------|--------------|------------|
+| [7 (alpha)][sequelize-core]     | >= 14                    | >= 4.4                    | ❓            | ❓          |
```
```suggestion
| [7 (alpha)][sequelize-core]     | ^14.17.0 || >= 16.0.0    | >= 4.4                    | ❓            | ❓          |
```

WikiRik commented on src/pages/releases.md:
```diff
@@ -0,0 +1,175 @@
+
+:::caution
+
+sqlite3 has not been released since february 2021, and has security vulnerabilities and installation issues on newer platforms.
```
```suggestion
sqlite3 has not been released since February 2021, and has security vulnerabilities and installation issues on newer platforms.
```

WikiRik reviewed (commented):
Few suggestions
